python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -351,6 +362,23 @@ def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any
     def get_autoscaling_actor_pools(self) -> List[AutoscalingActorPool]:
         return [self._actor_pool]
 
+    def _manage_actor_restarting_state(self, actor):
+        actor_state = actor._get_local_state()
+        if actor_state != gcs_pb2.ActorTableData.ActorState.ALIVE:
+            # If an actor is not ALIVE, it's a candidate to be marked as a
+            # restarting actor.
+            assert actor_state is gcs_pb2.ActorTableData.ActorState.RESTARTING
+            self._actor_pool.mark_running_actor_as_restarting(actor)
+        else:
+            # If an actor is ALIVE, it's a candidate to be marked as an
+            # alive actor, if not already the case.
+            self._actor_pool.mark_actor_as_alive(actor)
+
+    def update_resource_usage(self) -> None:
+        """Updates resources usage."""
+        for actor in self._actor_pool._running_actors.keys():
```

Comment:
Nit: Should we add a method to `_ActorPool` that provides a list of actor handles so that we don't access the internal `_running_actors` attribute?